### PR TITLE
[Don't Merge] upgrade sbt 1.9.8

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to upgrade sbt from `1.9.3` to `1.9.8`.

### Why are the changes needed?
After version sbt 1.9.3, if there are corrupt jars under the local (~/. m2), some workflows will fail to compile. Based on this, we added an environment variable (`SKIP_LOCAL_M2`) in SparkBuild to control whether to use the `Maven Local Resolver` mechanism.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: project/SparkBuild.scala [hunk 1]
```diff
@@ -254,6 +256,12 @@ object SparkBuild extends PomBuild {
     }
   )
 
+  lazy val cachePolicies = if (sys.env.contains("SKIP_LOCAL_M2")) {
```
Comment (LuciferYang) [inline project/SparkBuild.scala:259]: If this is necessary, this work should be completed in a separate PR first.

Comment (panbingkun) [reply project/SparkBuild.scala:259]: Okay, Let me try this solution first 😄

**Comment by panbingkun** [comment]
I changed my approach and modified some of the logic in `SparkBuild` by using an environment variable `SKIP_ M2_ LOCAL` is used to determine whether we `skip the local Maven local cache` (as some of our GA's corrupted jars in maven local cache have caused some of our workflows to fail).

Friendly ping @dongjoon-hyun, 
We would like to hear your suggestion. Is this current approach relatively feasible?

File: project/SparkBuild.scala [hunk 2]
```diff
@@ -264,16 +264,14 @@ object SparkBuild extends PomBuild {
...
     resolvers := Seq(
       // Google Mirror of Maven Central, placed first so that it's used instead of flaky Maven Central.
       // See https://storage-download.googleapis.com/maven-central/index.html for more info.
       "gcs-maven-central-mirror" at "https://maven-central.storage-download.googleapis.com/maven2/",
-      DefaultMavenRepository,
-      Resolver.mavenLocal,
-      Resolver.file("ivyLocal", file(Path.userHome.absolutePath + "/.ivy2/local"))(Resolver.ivyStylePatterns)
-    ),
+      DefaultMavenRepository) ++
+      { if (sys.env.contains("SKIP_LOCAL_M2")) Nil else Seq(Resolver.mavenLocal) } :+
```
Comment (panbingkun) [inline project/SparkBuild.scala:273]: Main modification points.

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -615,6 +615,7 @@ jobs:
       PYSPARK_DRIVER_PYTHON: python3.9
       PYSPARK_PYTHON: python3.9
       GITHUB_PREV_SHA: ${{ github.event.before }}
+      SKIP_LOCAL_M2: true
```
Comment (panbingkun) [inline .github/workflows/build_and_test.yml:618]: Only for `Linters, licenses, dependencies and documentation generation`

**Comment by panbingkun** [comment]
PTAL when you have some time, and Happy New Year 😄. @dongjoon-hyun

**Comment by dongjoon-hyun** [comment]
Thank you for your deep investigation, @panbingkun . Happy New Year.
For me, this approach is still a little reluctant because depending on skipping the local M2 is easily broken at various user environments.
Although we can make CI works here, it will bite us back harder when something is broken. It's because more difficult to debug.

**Comment by dongjoon-hyun** [comment]
If SBT does not provide any improvement, we may need to stick to the old SBT version.

**Comment by panbingkun** [comment]
> If SBT does not provide any improvement, we may need to stick to the old SBT version.

Okay, let's continue to observe `the next version`.

**Comment by panbingkun** [comment]
@dongjoon-hyun

has an offline discussion with @LuciferYang, another approach maybe:

Should we ignore `Maven repo cache` for `Mima tasks` in both `GA` and `developer environments`? 
so make it `not easily broken at various user environments`.

This bug in `coursier`, it is estimated that it will not be fixed for a while: https://github.com/coursier/coursier/issues/2942#issuecomment-2145674637
